hazelcast/src/main/java/com/hazelcast/config/AbstractCacheConfig.java

```diff
@@ -423,14 +430,23 @@ public boolean isStoreByValue() {
         return this;
     }
 
+    protected CacheConfiguration<K, V> setListenerConfigurations(Set<CacheEntryListenerConfiguration<K, V>> listeners) {
+        this.listenerConfigurations = DeferredValue.concurrentSetOfValues(listeners);
+        return this;
+    }
+
+    public Set<CacheEntryListenerConfiguration<K, V>> getListenerConfigurations() {
+        return DeferredValue.asPassThroughSet(listenerConfigurations, serializationService);
+    }
+
+    protected boolean hasListenerConfiguration() {
+        return listenerConfigurations != null && !listenerConfigurations.isEmpty();
+    }
+
     @Override
     public int hashCode() {
-        int result = cacheLoaderFactory != null ? cacheLoaderFactory.hashCode() : 0;
-        result = 31 * result + listenerConfigurations.hashCode();
```

Comment:
Can we return some kind of hash codes from the deferred values? So we keep these fields in the hashCode/equals?

Reply:
Yes, `DeferredValue` implements `hashCode`, so I will restore those fields in the `hashCode` calculation